mods/default/mapgen.lua

```diff
@@ -1655,9 +1754,18 @@ end
 
 
 --
--- Detect mapgen to select functions
+-- Detect mapgen, flags and parameters to select functions
 --
 
+-- Get setting or default
+local mgv7_spflags = minetest.get_mapgen_setting("mgv7_spflags") or
+	"mountains, ridges, nofloatlands"
+local captures_float = string.match(mgv7_spflags, "floatlands")
```

Comment:
It is definitely not good. It is evident from the hacky `elseif` you add below. There should be a more robust way to parse these flags.

Something like this should work:
```
local function parse_flags(str)
  local flags = { }
  for _, item in ipairs(str:split(",")) do
    item = item:trim()
    local negflag = item:match("^no(.*)")
    if negflag then
      flags[negflag] = false
    else
      flags[item] = true
    end
  end
  return flags
end
```

Then use something like:

```
local flag_floatlands = parse_flags(mgv7_spflags).floatlands == true
-- ...
elseif mg_name == "v7" and flag_floatlands then
-- ...
```